cfme/tests/containers/test_ssl_providers.py

```diff
@@ -0,0 +1,121 @@
+@pytest.mark.usefixtures('has_no_containers_providers')
+@pytest.mark.polarion('CMP-10586')
+def test_add_provider_ssl(provider, default_sec_protocols, soft_assert):
+    """ This test checks adding container providers with 3 different
+     security protocols in the Default tab only:
+    SSL trusting custom CA, SSL without validation and SSL
+        """
+    default_sec_protocols = ''.join(default_sec_protocols)
+    navigate_to(OpenshiftProvider, 'Add')
+    fill(provider.properties_form, form_data(provider, default_sec_protocol=default_sec_protocols))
+    for cred in provider.credentials:
+        try:
+            fill(provider.credentials[cred].form, provider.credentials[cred], validate=True)
+            provider._submit(False, provider.add_provider_button)
+            flash.assert_message_contain('Containers Providers "' + provider_name + '" was saved')
+        except:
+            # The refresh is a workaround for
+            # https://github.com/ManageIQ/integration_tests/issues/4478
+            sel.refresh()
+            soft_assert(False, provider_name + ' wasn\'t added successfully using ' +
+                        default_sec_protocols + ' security')
+
+
+@pytest.mark.parametrize(('default_sec_protocols', 'hawkular_sec_protocols'),
+                         product(DEFAULT_SEC_PROTOCOLS, HAWKULAR_SEC_PROTOCOLS))
+@pytest.mark.usefixtures('has_no_containers_providers')
+@pytest.mark.polarion('CMP-10586')
+def test_add_hawkular_provider_ssl(provider, default_sec_protocols,
+                                   hawkular_sec_protocols, soft_assert):
+    """This test checks adding container providers with 3 different security protocols in
+     Default and Hawkular tabs:
+    SSL trusting custom CA, SSL without validation and SSL
+        """
+    navigate_to(OpenshiftProvider, 'Add')
+    fill(provider.properties_form, form_data(provider, default_sec_protocol=default_sec_protocols,
+                                             hawkular_sec_protocol=hawkular_sec_protocols,
+                                             with_hawkular=True))
+    for cred in provider.credentials:
+        try:
+            fill(provider.credentials[cred].form, provider.credentials[cred], validate=True, validate_hawkular=True)
+            provider._submit(False, provider.add_provider_button)
+            flash.assert_message_contain('Containers Providers "' + provider_name + '" was saved')
+        except:
+            # The refresh is a workaround for
+            # https://github.com/ManageIQ/integration_tests/issues/4478
+            sel.refresh()
+            soft_assert(False, provider_name + ' wasn\'t added successfully using ' +
+                        default_sec_protocols + ' and ' +
+                        hawkular_sec_protocols + ' hawkular security protocol')
+
+
+def form_data(provider, default_sec_protocol, hawkular_sec_protocol=None, with_hawkular=False):
```

Comment:
honestly, I think you need to just construct provider object and then call provider.create() instead. 
You could make a fixture which would help you to setup provider object for tests according to some settings if you wish.